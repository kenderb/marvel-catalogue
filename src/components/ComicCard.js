import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { ComicCardWrapper } from './styled/lib';

const ComicCard = ({
  title, thumbnail, creators, id,
}) => {
  const thumbnailSplit = thumbnail.path.split('//');
  thumbnailSplit[0] = 'https://';

  const displayCreatorsName = () => (
    creators.items.map(creator => (
      <span key={creator.name}>
        {creator.name}
        ,
        {' '}
      </span>
    )));

  return (
    <ComicCardWrapper>
      <h2>{title}</h2>
      <img src={`${thumbnailSplit.join('')}.${thumbnail.extension}`} alt="" srcSet="" />
      <div>
        { creators.available > 0 ? displayCreatorsName() : '' }
      </div>
      <Link to={`/comics/${id}`}> See more </Link>
    </ComicCardWrapper>
  );
};

ComicCard.propTypes = {
  thumbnail: PropTypes.instanceOf(Object).isRequired,
  title: PropTypes.string.isRequired,
  creators: PropTypes.instanceOf(Object).isRequired,
  id: PropTypes.number.isRequired,
};

export default ComicCard;
