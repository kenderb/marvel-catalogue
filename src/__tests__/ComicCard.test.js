import { create } from 'react-test-renderer';
import ComicCard from '../components/Error';

test('ComicCard renders correctly', () => {
  const tree = create(<ComicCard />).toJSON();
  expect(tree).toMatchSnapshot();
});
