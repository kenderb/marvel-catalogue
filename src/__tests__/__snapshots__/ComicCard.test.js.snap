// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComicCard renders correctly 1`] = `
<h1>
  ERROR!!!
</h1>
`;
